collector_test.go

```diff
@@ -0,0 +1,535 @@
+package versionbundle
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/go-resty/resty"
+)
+
+func Test_Collector_Collect(t *testing.T) {
+	testCases := []struct {
+		HandlerFuncs    []func(w http.ResponseWriter, r *http.Request)
+		ExpectedBundles []Bundle
+	}{
+		// Test 0 ensures a single version bundle exposed by a single endpoint
+		// results in a single version bundle.
+		{
+			HandlerFuncs: []func(w http.ResponseWriter, r *http.Request){
+				func(w http.ResponseWriter, r *http.Request) {
+					cr := CollectorEndpointResponse{
+						VersionBundles: []Bundle{
+							{
+								Changelogs: []Changelog{
+									{
+										Component:   "calico",
+										Description: "Calico version updated.",
+										Kind:        "changed",
+									},
+								},
+								Components: []Component{
+									{
+										Name:    "calico",
+										Version: "1.1.0",
+									},
+									{
+										Name:    "kube-dns",
+										Version: "1.0.0",
+									},
+								},
+								Dependencies: []Dependency{},
+								Deprecated:   false,
+								Name:         "kubernetes-operator",
+								Time:         time.Unix(10, 5).In(time.UTC),
```

Comment:
That weirdness is actually necessary to get the tests working in circle and locally. I am not sure where a different time location is applied but the tests failed on circle before while passing on my laptop.